src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -404,10 +404,12 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
         else if (intrin.numOperands >= 2 && intrin.op2->IsEmbMaskOp())
         {
             // Handle case where op2 is operation that needs embedded mask
+            GenTree* op1 = intrin.op1;
             GenTree* op2 = intrin.op2;
             assert(intrin.id == NI_Sve_ConditionalSelect);
-            assert(op2->isContained());
+            assert(op1->OperIsHWIntrinsic());
```

Comment:
Can op1 not be a local?

What happens for something like a user explicitly writing `CndSel(mask, ..., ...)` where `mask` gets CSE'd, for example?

Reply:
ah fair. Removed it. The only place it is needed is in `IsMaskAllBitsSet()` which would return `false` if this is not `HWIntrinsic`.